src/System.Collections.NonGeneric/tests/Comparer/Comparer_DefaultInvariant.cs

```diff
@@ -7,6 +7,7 @@
 using System.Globalization;
 using Xunit;
 
+[Collection("DefaultThreadCurrentCulture")]
```

Comment:
I think I'd rather see parallelism disabled for this whole test suite rather than using a named collection like this.  None of these tests were written to be run in parallel, and I worry this is just one of potentially many such concurrency issues.  I think it'd make sense to disable parallelism for the library until it can be entirely rewritten (which it should be at some point).